src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -364,6 +375,37 @@ internal async Task<(IReadOnlyList<T> results, string continuationToken)> Execut
             return (results, page.ContinuationToken);
         }
 
+        private void UpdateSortIndex(FhirCosmosResourceWrapper cosmosWrapper)
+        {
+            var searchParameters = _supportedSearchParameters.Value.GetSearchParameters(cosmosWrapper.ResourceTypeName)
+                .Where(x => x.SortStatus != SortParameterStatus.Disabled)
+                .ToDictionary(x => x.Code);
+
+            if (searchParameters.Any())
+            {
+                foreach (KeyValuePair<string, SortValue> item in cosmosWrapper.SortValues.ToArray())
+                {
+                    if (!searchParameters.ContainsKey(item.Key))
+                    {
+                        cosmosWrapper.SortValues.Remove(item.Key);
+                    }
+                }
+
+                foreach (SearchParameterInfo field in searchParameters.Values)
+                {
+                    if (cosmosWrapper.SortValues.All(x => x.Value.SearchParameterUri != field.Url))
+                    {
+                        // Ensure sort property exists
+                        cosmosWrapper.SortValues.Add(field.Code, new SortValue());
```

Comment:
>cosmosWrapper.SortValues.Add(field.Code, new SortValue()); [](start = 24, length = 58)

nit: Would it be safer to add the Uri to this empty SortValue? I know we are not going to de-serialize the searchindexentry when reading resources. But I am afraid that somewhere in some change we are going to pass a FhirCosmosWrapper that has existing sort values and then this part will blow up because we already have an entry for the Code (and it was not caught because we were comparing only Uri).